examples/nlp/Tweet-classification-using-TFDF.py

```diff
@@ -0,0 +1,215 @@
+Last modified: 15/04/2022
+Description: Using Tensorflow Decision Forest for text classification
+"""
+"""
+## Introduction
+
+[TensorFlow Decision Forests](https://www.tensorflow.org/decision_forests) (TF-DF)
+provide powerful models, especially with structured data,
+It is a collection of state-of-the-art algorithms of Decision Forest models that are
+compatible with Keras APIs. The module includes Random Forests, Gradient Boosted Trees,
+and CART, and can be used for regression, classification, and ranking tasks.
+
+Alternatively for getting started
+In this example we will use Gradient Boosted Trees with pre-trained embeddings to
+classify disaster tweets.
+"""
+
+"""
+Install Tensorflow Decision Forest using following command : 
+`!pip3 install -q tensorflow_decision_forests`
+"""
+
+"""
+## Imports
+"""
+
+import pandas as pd
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+import tensorflow_hub as hub
+from tensorflow.keras import layers
+import tensorflow_decision_forests as tfdf
+import matplotlib.pyplot as plt
+
+"""
+## Get the data
+
+Dataset is avaliable at [kaggle](https://www.kaggle.com/c/nlp-getting-started)
+
+dataset description :
+
+1. Files
+
+    a. train.csv - the training set
+
+2. Columns
+
+    a. id - a unique identifier for each tweet
+
+    b. text - the text of the tweet
+
+    c. location - the location the tweet was sent from (may be blank)
+
+    d. keyword - a particular keyword from the tweet (may be blank)
+
+e. target - in train.csv only, this denotes whether a tweet is about a real disaster (1)
+or not (0)
+"""
+
+# Turn .csv files into pandas DataFrame's
+train_df = pd.read_csv(
+    "https://raw.githubusercontent.com/IMvision12/Tweets-Classification-NLP/main/train.csv"
+)
+print(train_df.head())
+
+# Printing shape of the training dataset
+print(train_df.shape)
+
+# Shuffling training data
+train_df_shuffled = train_df.sample(frac=1, random_state=42)
+# Dropping id, keyword and location columns as these columns consists of mostly nan values
+# we will be using only text and target columns
+train_df_shuffled.drop(["id", "keyword", "location"], axis=1, inplace=True)
+train_df_shuffled.reset_index(inplace=True, drop=True)
+print(train_df_shuffled.head())
+
+print(train_df_shuffled.info())
+
+# Printing total number of Disaster and non-Disaster tweets
+print(train_df_shuffled.target.value_counts())
+
+# Viewing 5 records from training data
+for ind, counter in enumerate(train_df_shuffled.index):
+    print(f"Target : {train_df_shuffled['target'][ind]}")
+    print(f"Text : {train_df_shuffled['text'][ind]}")
+    if counter == 5:
+        break
+
+# Splitting dataset into train and test
+test_df = train_df_shuffled.sample(frac=0.2, random_state=42)
+train_df = train_df_shuffled.drop(test_df.index)
+
+print(f"Using {len(train_df)} samples for training and {len(test_df)} for validation")
+
+"""
+## Convert data to tf.Dataset
+"""
+
+
+def create_dataset(dataframe, training=False):
```

Comment:
Optional:
What about using the function "pd_dataframe_to_tf_dataset". It is made specially to feed Pandas dataframes to 
 TF-DF  models.
https://www.tensorflow.org/decision_forests/api_docs/python/tfdf/keras/pd_dataframe_to_tf_dataset

In your case, the dataset is effectively consumed by the tf-hub embedding, but it should still work.

Reply:
I Tried `pd_dataframe_to_tf_dataset` earlier but i was getting errors :
![Capture](https://user-images.githubusercontent.com/88665786/163726044-412f886f-e464-4213-9acb-8e00f6ff0b91.PNG)